Crash when typing “he” in to field

Hi,

when I try to send an e-mail to a receiver who’s name starts with “he” the create new mail window closes, and I'm back on the K9 shared inbox. I can reproduce this since several weeks.
![Screenshot_2021-11-05-14-01-01-940_com fsck k9](https://user-images.githubusercontent.com/6573464/140515601-360465e9-0353-4d2d-91c5-e303fbfcccf4.jpg)

K9 Version: 5.806
Android Version: 11 RP1A200720.012
Xiaomi MIUI Version: 12.5.3

Yours

Roland

Thanks for the debug log :+1:

Exception from the log:
```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at androidx.loader.content.ModernAsyncTask$3.done(ModernAsyncTask.java:164)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)
Caused by: java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.util.TimSort.mergeHi(TimSort.java:899)
	at java.util.TimSort.mergeAt(TimSort.java:516)
	at java.util.TimSort.mergeCollapse(TimSort.java:441)
	at java.util.TimSort.sort(TimSort.java:245)
	at java.util.Arrays.sort(Arrays.java:1492)
	at java.util.ArrayList.sort(ArrayList.java:1470)
	at java.util.Collections.sort(Collections.java:206)
	at com.fsck.k9.activity.compose.RecipientLoader.fillContactDataFromQuery(RecipientLoader.java:342)
	at com.fsck.k9.activity.compose.RecipientLoader.loadInBackground(RecipientLoader.java:195)
	at com.fsck.k9.activity.compose.RecipientLoader.loadInBackground(RecipientLoader.java:37)
	at androidx.loader.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:307)
	at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:60)
	at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:48)
	at androidx.loader.content.ModernAsyncTask$2.call(ModernAsyncTask.java:141)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
```

--- a/app/ui/legacy/src/main/java/com/fsck/k9/activity/compose/RecipientLoader.java
+++ b/app/ui/legacy/src/main/java/com/fsck/k9/activity/compose/RecipientLoader.java
@@ -110,8 +110,12 @@ public class RecipientLoader extends AsyncTaskLoader<List<Recipient>> {
                 return timesContactedDiff;
             }
 
-            if (lhs.sortKey == null || rhs.sortKey == null) {
+            if (lhs.sortKey == null && rhs.sortKey == null) {
                 return 0;
+            } else if (lhs.sortKey == null) {
+                return 1;
+            } else if (rhs.sortKey == null) {
+                return -1;
             }
 
             return CASE_INSENSITIVE_ORDER.compare(lhs.sortKey, rhs.sortKey);

